### PR TITLE
Silence failing tests on Windows and add Windows build step

### PR DESCRIPTION
**Pull request by tagniam**
**What this PR does / why we need it**:
While working on the bang/filter command feature, I've noticed that quite a few tests are failing only on Windows. This PR proposes we temporarily silence those tests on Windows and document them with TODO comments, which I hope should be helpful towards fixing #4844 (and #4079).

Additionally, I propose we add a Windows build step to our Github Actions workflow to encourage cross-platform compatibility for future PRs. However, this may affect this repo's Github Actions minutes limit - [jobs that run on Windows consume minutes at twice the rate of Linux-based jobs](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#:~:text=Jobs%20that%20run%20on%20Windows%20and,10%2C000%20minutes%20included%20in%20your%20account.). This multiplier means that this would probably cause the minutes consumption to triple. I'm not sure what the minutes limit is for VSCodeVim nor if we currently run near that limit, so if this would cause problems, I can omit this change. Lemme know your thoughts!

**Which issue(s) this PR fixes**
N/A, but progresses towards fixing #4844 and #4079.

### CODE REVIEW

File: test/jumpTracker.test.ts [hunk 1]
```diff
@@ -358,19 +369,22 @@ suite('Record and navigate jumps', () => {
...
-      newJumpTest({
+      // TODO: this fails on Windows
+      newJumpTestSkipOnWindows({
         start: ['|start', 'a1', 'a2', 'a3', 'a4', 'a5', 'end'],
         keysPressed: '/^\nnnnkdd',
         end: ['start', 'a1', 'a2', '|a4', 'a5', 'end'],
         jumps: ['start', 'a1', 'a2', 'a4'],
       });
-      newJumpTest({
+      // TODO: this fails on Windows
```
Comment (J-Fields) [inline test/jumpTracker.test.ts:386]: Can you make these `// TODO(#4844): this fails on Windows` so it's associated with the right ticket?

**Comment by J-Fields** [comment]
Thanks for the PR! I'm definitely supportive of this idea. We should do something similar for the web version as well at some point (maybe Mac as well, but it seems that uses 10x the minutes).

Any idea why this last step won't finish? 
image

> I'm not sure what the minutes limit is for VSCodeVim nor if we currently run near that limit

I think the free limit is 2000 minutes/month - I don't think we're anywhere near that. If this causes issues, we can always change to run on all platforms only for releases, or something like that.

File: test/jumpTracker.test.ts (hunk 1 above)
Comment (tagniam) [reply test/jumpTracker.test.ts]: Done!

**Comment by tagniam** [comment]
> I think the free limit is 2000 minutes/month - I don't think we're anywhere near that. If this causes issues, we can always change to run on all platforms only for releases, or something like that.

Sounds good!

> Any idea why this last step won't finish?

Not sure, the other two steps are fine :thinking: I've tried closing/reopening the PR, force pushing and even changing the base branch. Going to try and open another PR to see if this is a fluke.

**Comment by tagniam** [comment]
Sorry for the delay! From what I can tell, the last `build` step is forever waiting because it comes from the existing enforced status check rule - it is applying to this branch as well, but since the rule was updated and no longer exists on this branch, it's waiting forever.

You might need to merge this in first and then modify the required status checks in this repo's `Settings > Branches > Require status checks to pass before merging` to these new `build / build (ubuntu-latest)` and `build / build (windows-latest)` checks. Related: https://circleci.com/docs/2.0/enable-checks/#troubleshooting-github-checks-waiting-for-status-in-github

---
One thing I'd like to bring up is that it seems running tests on Windows is a bit flakier, sometimes the Windows build fails while the ubuntu build succeeds, in which a re-run usually fixes it. If that is a concern we can shelve the Windows build for now, or perhaps make it a non-required check for PRs. Thoughts?

**Comment by J-Fields** [comment]
> Sorry for the delay!

No trouble whatsoever, I haven't been active on this project at all recently.

> You might need to merge this in first and then modify the required status checks

Sounds good, I'll give that a shot

> One thing I'd like to bring up is that it seems running tests on Windows is a bit flakier, sometimes the Windows build fails while the ubuntu build succeeds, in which a re-run usually fixes it. If that is a concern we can shelve the Windows build for now, or perhaps make it a non-required check for PRs. Thoughts?

Yeah good point. Non-required is probably the way to go for now

**Comment by J-Fields** [comment]
Seems to have worked, thanks very much!

**Pull request merged**